relay-statsd/src/lib.rs

```diff
@@ -162,6 +161,26 @@ pub fn set_client(client: MetricsClient) {
     *METRICS_CLIENT.write() = Some(Arc::new(client));
 }
 
+/// Set a test client for the period of the called function.
+/// Note: This also changes the METRICS_CLIENT for other threads.
+pub fn with_capturing_test_client(f: impl FnOnce()) -> Vec<String> {
+    let old_client = METRICS_CLIENT.read().clone();
```

Comment:
I don't think this is safe. Tests run on multiple threads in the same process.

Why not temporarily set the thread local instead?

Reply:
Done.